c++/src/TypeImpl.cc

```diff
@@ -498,54 +506,54 @@ namespace orc {
   }
 
   ORC_UNIQUE_PTR<Type> Type::buildTypeFromString(const std::string& input) {
-    std::vector<std::pair<std::string, Type*> > res =
+    std::vector<std::pair<std::string, ORC_UNIQUE_PTR<Type> > > res =
       TypeImpl::parseType(input, 0, input.size());
     if (res.size() != 1) {
       throw std::logic_error("Invalid type string.");
     }
-    return ORC_UNIQUE_PTR<Type>(res[0].second);
+    return std::move(res[0].second);
   }
 
   Type* TypeImpl::parseArrayType(const std::string &input,
```

Comment:
is it better to change return type to unique_ptr as well to be consistent? same for below.